test/sql/types/enum/test_enum_from_query.test

```diff
@@ -0,0 +1,148 @@
+# description: ENUM tests
+# group: [enum]
+
+statement ok
+PRAGMA enable_verification
+
+statement error
+CREATE TYPE type_a AS ENUM (SELECT '1', '2');
+
+statement error
+CREATE TYPE type_a AS ENUM (SELECT 1);
+
+statement ok
+CREATE TABLE test (x VARCHAR, y VARCHAR);
+
+statement error
+CREATE TYPE a AS ENUM (SELECT * FROM test);
+
+statement ok
+DROP TABLE test;
+
+statement ok
+CREATE TABLE number_str AS (SELECT CAST(range AS VARCHAR) AS x FROM range(1, 100));
+
+statement ok
+CREATE TYPE number_enum AS ENUM (SELECT x FROM number_str WHERE x >= '50');
+
+statement ok
+CREATE TABLE test (x number_enum);
+
+statement error
+INSERT INTO test VALUES ('49');
+
+statement ok
+INSERT INTO test SELECT CAST(range AS VARCHAR) FROM range(50, 100);
+
+statement ok
+INSERT INTO test VALUES ('6'), ('7'), ('8'), ('9');
+
+statement ok
+INSERT INTO test SELECT CAST(range AS VARCHAR) FROM range(50, 100);
+
+statement ok
+DROP TABLE number_str;
+
+statement ok
+DROP TABLE test;
+
+statement ok
+DROP TYPE number_enum;
+
+
+# Test multiple vector
+statement ok
+CREATE TABLE number_str AS (SELECT CAST(range AS VARCHAR) AS x FROM range(1, 4000));
+
+statement ok
+CREATE TYPE number_enum AS ENUM (SELECT x FROM number_str WHERE x != '500' AND x != '1500');
+
+statement ok
+CREATE TABLE test (x number_enum);
+
+statement ok
+INSERT INTO test SELECT CAST(range AS VARCHAR) FROM range(1, 500);
+
+statement ok
+INSERT INTO test SELECT CAST(range AS VARCHAR) FROM range(501, 1500);
+
+statement ok
+INSERT INTO test SELECT CAST(range AS VARCHAR) FROM range(1501, 4000);
+
+statement error
+INSERT INTO test VALUES ('500');
+
+statement error
+INSERT INTO test VALUES ('1500');
+
+statement ok
+DROP TABLE number_str;
+
+statement ok
+DROP TABLE test;
+
+statement ok
+DROP TYPE number_enum;
+
+# Test NULL handing
+statement ok
+CREATE TABLE number_str AS (SELECT CAST(range AS VARCHAR) AS x FROM range(0, 64));
+
+statement ok
+INSERT INTO number_str SELECT NULL FROM range(0, 64);
+
+statement ok
+INSERT INTO number_str SELECT CAST(range AS VARCHAR) FROM range(64, 96);
+
+statement ok
+INSERT INTO number_str SELECT NULL FROM range(0, 64);
+
+statement ok
+INSERT INTO number_str SELECT CAST(range AS VARCHAR) FROM range(96, 100);
+
+statement ok
+CREATE TYPE number_enum AS ENUM (SELECT x FROM number_str);
+
+statement ok
+CREATE TABLE test (x number_enum);
+
+statement ok
+INSERT INTO test SELECT CAST(range AS VARCHAR) FROM range(0, 100);
+
+statement ok
+DROP TABLE number_str;
+
+statement ok
+DROP TABLE test;
+
+statement ok
+DROP TYPE number_enum;
+
+# Test NULL handing
+statement ok
+CREATE TABLE number_str (x VARCHAR);
+
+statement ok
+INSERT INTO number_str VALUES ('1'), ('2'), (NULL), ('3'), (NULL), (NULL), ('4');
+
+statement ok
+CREATE TYPE number_enum AS ENUM (SELECT x FROM number_str);
+
+statement ok
+CREATE TABLE test (x number_enum);
+
+statement ok
+INSERT INTO test SELECT CAST(range AS VARCHAR) FROM range(1, 5);
```

Comment:
Can we add the following tests:

* Creating an enum from a query with duplicate values (which should succeed, and create an enum with distinct values)
* Create an enum from a query with an empty result set (this should throw an error)
* Add tests where we create differently sized enums (100 rows, 10000 rows, 100000 rows)
* Add a slow test where the number of unique values exceeds the maximum number of values in an enum (`2^32` values, i.e. `4294967296` values). This should fail as well.